src/App.tsx

```diff
@@ -1,60 +1,109 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
-import classNames from 'classnames';
+import cn from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getPosts, getUsers } from './services/api';
 
 export const App: React.FC = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isPostsLoading, setIsPostsLoading] = useState(false);
+
+  const [usersFromServer, setUsersFromServer] = useState<User[] | null>(null);
+  const [postsFromServer, setPostsFromServer] = useState<Post[] | null>(null);
+
+  const [isErrorShowing, setIsErrorShowing] = useState(false);
+
+  const selectUser = (user: User) => {
+    setSelectedUser(user);
+    setSelectedPost(null);
+
+    setIsPostsLoading(true);
+    getPosts(user.id, setIsErrorShowing, setPostsFromServer)
```

Comment:
I meant only API calls, set posts and errors you can do inside a component in this case.

The same logic as in the example
```typescript
const getUsers = () => {
    return client.get<User[]>('/users');
};
```

If you need clarification don't hesitate to ask any questions in the Q&A chat